### ISSUE TITLE
Model Loss, optimal epoch and other model metrics

### ISSUE THREAD

**Issue by pythonometrist** [ThilinaRajapakse/simpletransformers#754]
**Is your feature request related to a problem? Please describe.**
It would be great to return some standard metrics at the end of training - epoch the training stopped, the loss (either training or eval), mcc, etc. optimizer state.

**Describe the solution you'd like**
A simple dictionary of metrics required. The model evaluates these measures/tracks them  in each iteration in any case.

**Describe alternatives you've considered**
At the moment I dont see any easy way to extract the loss, number of epochs. For example if i want to run 5 fold cross validation and figure out which epoch the earlystopping picked in each fold (and use the average of that for training a full model) I cant see an easy way to extract that info.

**Additional context**

**Comment by ThilinaRajapakse** [comment]
The `train_model()` methods now return `global_step, training_details`.

global_step: Number of global steps trained
training_details: Average training loss if evaluate_during_training is False or full training progress scores if evaluate_during_training is True
